src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -21670,12 +21670,12 @@ public void InlineArrayElement_01()
                 comp1,
                 verify: ExecutionConditionUtil.IsMonoOrCoreClr ? Verification.Passes : Verification.Skipped).
             VerifyDiagnostics(
-                // (6,7): warning CS9184: 'Inline arrays' language feature is not supported for an inline array type that is not valid as a type argument, or has element type that is not valid as a type argument.
-                //     T _f;
-                Diagnostic(ErrorCode.WRN_InlineArrayNotSupportedByLanguage, "_f").WithLocation(6, 7),
-                // (12,7): warning CS9184: 'Inline arrays' language feature is not supported for an inline array type that is not valid as a type argument, or has element type that is not valid as a type argument.
-                //     S _f;
-                Diagnostic(ErrorCode.WRN_InlineArrayNotSupportedByLanguage, "_f").WithLocation(12, 7)
+                // (3,12): warning CS9184: 'Inline arrays' language feature is not supported for an inline array type that is not valid as a type argument, or has element type that is not valid as a type argument.
```

Comment:
>warning CS9184: 'Inline arrays' language feature is not supported for an inline array type that is not valid as a type argument, or has element type that is not valid as a type argument. [](http://example.com/codeflow?start=27&length=186)

Is this warning correct? Couldn't `S1<T>` be used as a type argument a type `C<T> where T : allows ref struct`?

Reply:
> Is this warning correct? 

In a general case a `ref` struct cannot be used as a type argument. That is the case the warning refers to. If that will be confusing, we will be able to adjust the wording in the future.